pandas/core/internals/blocks.py

```diff
@@ -475,7 +475,12 @@ def coerce_to_target_dtype(self, other, warn_on_upcast: bool = False) -> Block:
             and isna(other)
             and other is not NaT
         ):
-            warn_on_upcast = False
+            try:
+                is_np_nat = np.isnat(other)
+            except TypeError:
+                is_np_nat = False
```

Comment:
I think `isinstance(other, (np.datetime64, np.timedelta64))` is more explicit and avoids a try/except?